prime-angular-frontend/src/app/shared/components/college-certification-form/college-certification-form.component.ts

```diff
@@ -122,8 +124,19 @@ export class CollegeCertificationFormComponent implements OnInit {
       });
 
     this.licenseCode.valueChanges
-      .subscribe((licenseCode: number) =>
+      .subscribe((licenseCode: number) => {
+
+        const matchingLicenses = this.licenses.filter(licenseConfig => licenseConfig.code === licenseCode);
+        if (matchingLicenses.length != 1) {
+          this.formUtilsService.setValidators(this.licenseNumber, [Validators.required, FormControlValidators.alphanumeric]);
+        } else if ((matchingLicenses[0] as LicenseWeightedConfig).validate) {
+          // For college licences that we validate against the pharmanet tables, restrict the licence # to 5 digits
+          this.formUtilsService.setValidators(this.licenseNumber, [Validators.required, FormControlValidators.numeric, FormControlValidators.requiredLength(5, 5)]);
+        } else {
+          this.formUtilsService.setValidators(this.licenseNumber, [Validators.required, FormControlValidators.alphanumeric]);
+        }
```

Comment:
When I entered exact 5 alphabets, the validation only highlighted red but no error prompt message. Would it also be intuitive to tell user "College license number needs to be 5 digits"?

<img width="697" alt="Capture" src="https://user-images.githubusercontent.com/58282608/102831862-a4496080-43a1-11eb-9862-4837bd1a4889.PNG">

Reply:
Good catch!  I fixed this error.  Please review again Johnny.